src/builder.ts

```diff
@@ -1,28 +1,31 @@
 import * as Unist from 'unist';
 import { IRoot } from './ast-types/mdast';
 import { Reader } from './reader';
-import { ILangReader } from './reader/types';
 import { stringify } from './stringify';
 
 export class Builder {
   public root: IRoot;
 
-  constructor(public reader: ILangReader = new Reader()) {
+  constructor(public reader = new Reader()) {
```

Comment:
Suggest reverting, as right now, you will always need to pass an instance of `Reader`, rather than an object implementing `ILangReader` interface.
Basically the following won't satisfy TS.
```ts
     const x: ILangReader = {
        toLang(data: IRoot): string {
        },
        fromLang(raw: string): IRoot {
          return {} as any;
        }
      }
      const b = new Builder(x);
```

Reply:
Ah, I see now. `ILangReader` does not have `fromSpec` and `toSpec` methods. Fair enough.